src/app/components/food/generator/FoodModal/ListIngredients/index.js

```
@@ -0,0 +1,37 @@
+import { authProvider } from "@/app/provider/authProvider/clientSide";
+import ProvideToken from "@/app/provider/sessionProvider";
+import React,{ useEffect, useState } from "react"
+
+function ListIngredientsUI({ ingredients }) {
+  console.log(ingredients)
+    return (
+        <ul class="list-disc list-inside text-[--color-white] font-semibold">
+          {ingredients && ingredients.map((i) => (
+            <li>{i.name}</li>
+          ))}
+        </ul>
+    )
+}
+
+function ListIngredientsLogique({UI, meal, token}){
+  const [recipe, setRecipe] = useState(null);
+  useEffect(()=>{
+    authProvider.getRecipe(meal.recipe.id,token).then((recipe)=>{
+        setRecipe(recipe);
+    }).catch((e)=>{
+        console.log("can't fetch recipe")
```

return the error instead of logging it.